src/components/shared/Deeplinking.tsx

```diff
@@ -0,0 +1,22 @@
+import { NavigationActions } from "react-navigation";
+import url from "url";
+import { DEEPLINK_ROUTES } from "./Navigation";
```

Comment:
will do!

Reply:
Actually, looks like I can't pull this out without also pulling out RouteName as well. Possibly due to circular dependency loading (need to import Deeplinking in Navigation and Navigation in Deeplinking). Since that change would touch a lot of files, I'll just leave a TODO.